benches/cub/device/select/if.cu

```diff
@@ -0,0 +1,176 @@
+#include <nvbench/nvbench.cuh>
+
+#include <thrust/device_vector.h>
+#include <thrust/sequence.h>
+// Why is this in detail?
+#include <thrust/detail/raw_pointer_cast.h>
+
+#include <cub/device/device_select.cuh>
+
+#include <tbm/range_generator.cuh>
+
+enum class select_op_type
+{
+  greater_than_middle,
+  greater_than_zero,
+  even
+  // complex
+};
+
+template <typename T>
+struct gt_select_op
+{
+  T m_val {};
+
+  explicit gt_select_op(T val)
+      : m_val(val)
+  {}
+
+  __device__ bool operator()(const T& val)
+  {
+    return val > m_val;
+  }
+};
+
+template <typename T>
+struct even_select_op
+{
+  __device__ bool operator()(const T& val)
+  {
+    return (val % 2) == 0;
+  }
+};
+
+template <select_op_type op_type>
+struct op_construction_helper;
+
+template <>
+struct op_construction_helper<select_op_type::greater_than_middle>
+{
+  template <typename T>
+  static gt_select_op<T> create_select_op(int elements)
+  {
+    return gt_select_op<T> (elements / 2);
```

Comment:
Remove the space and prefer `{}` to clarify that this is construction and not a call.